pennylane/fermi/conversion.py

```diff
@@ -148,3 +149,139 @@ def _(fermi_operator: FermiSentence, ps=False, wire_map=None, tol=None):
         return qubit_operator.map_wires(wire_map)
 
     return qubit_operator
+
+
+def parity_transform(
+    fermi_operator: (Union[FermiWord, FermiSentence]),
+    n: int,
+    ps: bool = False,
+    wire_map: dict = None,
+    tol: float = None,
+) -> Union[Operator, PauliSentence]:
+    r"""Convert a fermionic operator to a qubit operator using the parity mapping.
+    In parity mapping, qubit :math:`j` stores the parity of all :math:`j-1` qubits before it
+    whereas in Jordan-Wigner, qubit :math:`j` stores the occupation number of qubit :math:`j`.
+    The fermionic creation and annihilation operators are mapped to the Pauli operators as
+
+    .. math::
+        \begin{align*}
+           a^{\dagger}_0 &= \left (\frac{X_0 - iY_0}{2}  \right )\otimes X_1 \otimes X_2 \otimes ... X_n, \\\\
+           a^{\dagger}_n &= \left (\frac{Z_{n-1} \otimes X_n - iY_n}{2} \right ) \otimes X_{n+1} \otimes X_{n+2} \otimes ... \otimes X_n
+        \end{align*}
+
+    and
+
+    .. math::
+        \begin{align*}
+           a_0 &= \left (\frac{X_0 + iY_0}{2}  \right )\otimes X_1 \otimes X_2 \otimes ... X_n,\\\\
+           a_n &= \left (\frac{Z_{n-1} \otimes X_n + iY_n}{2} \right ) \otimes X_{n+1} \otimes X_{n+2} \otimes ... \otimes X_n
+        \end{align*}
+
+    where :math:`X`, :math:`Y`, and :math:`Z` are the Pauli operators and :math:`n` is the number of qubits, i.e., spin orbitals.
+
+    Args:
+        fermi_operator(FermiWord, FermiSentence): the fermionic operator
+        n (int): Number of qubits, i.e., spin orbitals in the system
+        ps (bool): whether to return the result as a PauliSentence instead of an
+            Operator. Defaults to False.
+        wire_map (dict): a dictionary defining how to map the orbitals of
+            the Fermi operator to qubit wires. If None, the integers used to
+            order the orbitals will be used as wire labels. Defaults to None.
+        tol (float): tolerance for discarding the imaginary part of the coefficients
+
+    Returns:
+        Union[PauliSentence, Operator]: a linear combination of qubit operators
+
+    **Example**
+
+    >>> w = FermiWord({(0, 0) : '+', (1, 1) : '-'})
+    >>> parity_transform(w, n=6)
+    (-0.25j*(PauliY(wires=[0]))) + ((-0.25+0j)*(PauliX(wires=[0]) @ PauliZ(wires=[1]))) +
+    ((0.25+0j)*(PauliX(wires=[0]))) + (0.25j*(PauliY(wires=[0]) @ PauliZ(wires=[1])))
+
+    >>> parity_transform(w, n=6, ps=True)
+    -0.25j * Y(0)
+    + (-0.25+0j) * X(0) @ Z(1)
+    + (0.25+0j) * X(0)
+    + 0.25j * Y(0) @ Z(1)
+
+    >>> parity_transform(w, n=6, ps=True, wire_map={0: 2, 1: 3})
+    -0.25j * Y(2)
+    + (-0.25+0j) * X(2) @ Z(3)
+    + (0.25+0j) * X(2)
+    + 0.25j * Y(2) @ Z(3)
+    """
+    warnings.warn("This mapping is not compatible with other functions in PennyLane", UserWarning)
+
+    return _parity_transform_dispatch(fermi_operator, n, ps, wire_map, tol)
```

Comment:
With this Hamiltonian:

```
symbols = ["H", "H"]   
geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.3889487]], requires_grad=False) 
mol = qml.qchem.Molecule(symbols, geometry) 
H = qml.qchem.fermionic_hamiltonian(mol)()
qml.fermi.parity_transform(H, 4, ps=True)

```

I get terms such as 

`0j * Z(1) @ Y(2) @ Y(0)`

in the final output. Could you inspect and see how this terms can be removed? 

(comparing with `jordan_wigner` might help fixing the issue)

Reply:
I think jordan_wigner has the same issue, I have changed it for parity_transform, do you need me to change it for jordan_wigner as well?

Reply:
> do you need me to change it for jordan_wigner as well

No, we can do it in a separate PR. Could you push the change for fixing it for parity_transform?